Makefile

```diff
@@ -15,15 +15,22 @@ clean:
 	@rm -f *.sqlite
 	@rm -rf .cache
 
-init:
```

Comment:
We can change, but my proposal is the one in code, which means:
The `make init` install the project so you can use it, all the dependencies needed in the modules.
And the `make init-dev` install all the usage and extras the developer needs to work on it.

So `make init` for users and `make init-dev` for developers

Reply:
Not because of the way poetry works with `extras`.
You have to specify the extras you are installing, otherwise it will even remove the other ones listed on the toml file.

So if you do:
`poetry install --extras 'tests' --extras 'dev' --extras 'docs'`

This will install every dependency in the toml file.

And later  do:
`poetry install`

This will reinstall all the "vanila" dependencies and will remove all the extras dependencies.

Reply:
In a nutshell, this was to way to make it different from user to dev

Reply:
think we need to step back and define the roles and goals


it feels like we are discussing 3 different make targets, and we should agree on what and why they do first, and not derive that meaning from the current or historic implementation?

1. `init` --> update all core deps  (for who for what is this useful?)
2. `init-dev` --> include the extra deps for tests, docs, ...  (for devs and robots doing CI)
3. `install` --> make a local install of the resulting package in the current virtualenv?  (for users wanting to use the core results in their env)

Reply:
```
startup:
	pip install --upgrade pip
	which poetry >/dev/null || pip install poetry

init: startup
       poetry install

init-dev: startup
	poetry install --extras 'tests' --extras 'dev' --extras 'docs'
	poetry run pre-commit install
	poetry run pre-commit install --hook-type commit-msg

init-docs: startup
	poetry install --extras 'docs'
```

Would this be good then?

1.  I would keep this for case of only package usage.
2. Done
3. Removed

Reply:
good to present the overview

I don't feel the need for init-docs but that might be just me ?

Reply:
This way the page builder does not need to have the full on install, just to update or make documentation.
That is the idea behind it